src/qforte/ucc/spqe.py

```diff
@@ -178,12 +181,13 @@ def run(self,
                             sq_op.add(-1.0, occ_idx[::-1], unocc_idx[::-1])
                             sq_op.simplify()
                             self._pool_obj.add_term(0.0, sq_op)
-                            self._excitation_dictionary[I] = idx
+                            self._coeff_idx_to_pool_idx[I] = idx
                             self._indices_of_zeroable_residuals_for_pool.add(I)
                             idx += 1
 
-        # excitation operator index : determinant id
-        self._reversed_excitation_dictionary = {value: key for key, value in self._excitation_dictionary.items()}
+        # Given a pool index, what is the coefficient of the "corresponding" coefficient vector element? Used to extract significant residuals in microiterations.
+        # WARNING! This variable will likely need to be replaced to support generalized pools.
+        self._pool_idx_to_coeff_idx = {value: key for key, value in self._coeff_idx_to_pool_idx.items()}
```

Comment:
I am not sure why the variable would need to be replaced. Even for generalized excitations, one should still have a one-to-one correspondence between excitation operators and projections. The trouble would be assigning the generalized excitations to particular determinants, since now there is some freedom. A possible trouble with GSD and PQE, in particular when using minimum basis sets, is that the operator pool is larger than FCI. That would make PQE UCCGSD impossible. unless you select a subset of generalized excitations or come up with a different optimization procedure.